src/commands/commerce/extension/map.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { flags, SfdxCommand } from '@salesforce/command';
+import { Messages, Org, SfdxError } from '@salesforce/core';
+import { forceDataSoql, forceDataRecordCreate } from '../../../lib/utils/sfdx/forceDataSoql';
+import { StatusFileManager } from '../../../lib/utils/statusFileManager';
+
+Messages.importMessagesDirectory(__dirname);
+
+const TOPIC = 'extension';
+const CMD = `commerce:${TOPIC}:map`;
+const msgs = Messages.loadMessages('@salesforce/commerce', 'store');
+
+export class MapExtension extends SfdxCommand {
+    public static readonly requiresUsername = true;
+
+    public static description = msgs.getMessage('extension.map.cmdDescription');
+    public static example = [`sfdx ${CMD} --registered-extension-name --store-id --store-name`];
+    public static flagsConfig = {
+        'registered-extension-name': flags.string({
+            char: 'r',
+            description: msgs.getMessage('extension.map.regExtensionNameFlagDescription'),
+        }),
+        'store-name': flags.string({
+            char: 'n',
+            description: msgs.getMessage('extension.map.StoreNameFlagDescription'),
+        }),
+        'store-id': flags.string({
+            char: 'i',
+            description: msgs.getMessage('extension.map.storeId'),
+        }),
+    };
+    public org: Org;
+    public statusFileManager: StatusFileManager;
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async run(): Promise<string> {
+        this.ux.log(`Accessing Store using username: ${this.org.getUsername()} \n..........`);
+
+        return this.processMapExtension(
+            this.flags['registered-extension-name'],
+            this.flags['store-name'],
+            this.flags['store-id'],
+            this.org.getUsername()
+        );
+    }
+
+    public processMapExtension(extensionName: string, storeName: string, storeId: string, userName: string): string {
+        const storeid = this.getStoreId(storeName, storeId, userName);
+        const registeredExternalServiceId = this.getExtensionName(extensionName, userName);
+
+        if (storeName === undefined && storeId === undefined) {
+            throw new SfdxError(msgs.getMessage('extension.map.undefinedName'));
+        }
```

Comment:
I believe this condition needs to be above line 54 otherwise it will never get run because `getStoreId()` will throw an error for errStoreName or errStoreId